File: jambeth/jAmbeth.Cache.DataChange/src/de/osthus/ambeth/cache/ServiceResultCacheDCL.java
package de.osthus.ambeth.cache;

import de.osthus.ambeth.datachange.IDataChangeListener;
import de.osthus.ambeth.datachange.model.IDataChange;
import de.osthus.ambeth.ioc.annotation.Autowired;
import de.osthus.ambeth.log.ILogger;
import de.osthus.ambeth.log.LogInstance;

public class ServiceResultCacheDCL implements IDataChangeListener
{
	@SuppressWarnings("unused")
	@LogInstance
	private ILogger log;

	@Autowired
	protected IServiceResultCache serviceResultCache;

	@Override
	public void dataChanged(IDataChange dataChange, long dispatchTime, long sequenceId)
	{
		if (dataChange.isEmpty())
		{
			return;
		}
		serviceResultCache.invalidateAll();
	}
}

File: ambeth/Ambeth.CacheDataChange/ambeth/cache/ServiceResultCacheDCL.cs
using System;
using De.Osthus.Ambeth.Datachange;
using De.Osthus.Ambeth.Datachange.Transfer;
using System.Collections.Generic;
using De.Osthus.Ambeth.Merge.Transfer;
using De.Osthus.Ambeth.Util;
using De.Osthus.Ambeth.Typeinfo;
using System.Collections;
using De.Osthus.Ambeth.Log;
using System.Threading;
using De.Osthus.Ambeth.Merge.Model;
using De.Osthus.Ambeth.Datachange.Model;
using De.Osthus.Ambeth.Ioc;

namespace De.Osthus.Ambeth.Cache
{
    public class ServiceResultCacheDCL : IDataChangeListener, IInitializingBean
    {
        [LogInstance]
        public ILogger Log { private get; set; }

        public IServiceResultCache ServiceResultCache { get; set; }

        public virtual void AfterPropertiesSet()
        {
            ParamChecker.AssertNotNull(ServiceResultCache, "ServiceResultCache");
        }

        public virtual void DataChanged(IDataChange dataChange, DateTime dispatchTime, long sequenceId)
        {
            if (dataChange.IsEmpty)
            {
                return;
            }
            ServiceResultCache.InvalidateAll();
        }
    }
}